stacktrace.go

```diff
@@ -125,6 +125,12 @@ func (st *stacktrace) Next() (_ runtime.Frame, more bool) {
 	return st.frames.Next()
 }
 
+// TakeStacktrace returns the stack trace,
+// and skips the provided number of frames.
+func TakeStacktrace(skip int) string {
```

Comment:
Should we `deprecate` this and add a comment that it's just a temporary export?

Reply:
I don't like the idea of exporting this functionality from the main Zap package even as a temporary unsupported thing.
Instead, I've moved the relevant bits into an internal package in https://github.com/uber-go/zap/pull/1341.

That makes it easy to use them here.

Reply:
Thanks for this i agree on this approach too